### PR TITLE
Added `at_height` and `absolute_timelocks`

### PR DESCRIPTION
**Pull request by SarcasticNastik**
Related to #247  and #248 . Added the `at_height` function for analyzing policies and `absolute_timelocks` for semantic analysis.

### CODE REVIEW

File: src/policy/semantic.rs [hunk 1]
```diff
@@ -499,6 +499,34 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
...
+    pub fn real_absolute_timelocks(&self) -> Vec<u32> {
+        match *self {
+            Policy::Unsatisfiable
+            | Policy::Trivial
+            | Policy::KeyHash(..)
+            | Policy::Sha256(..)
+            | Policy::Hash256(..)
+            | Policy::Ripemd160(..)
+            | Policy::Hash160(..) => vec![],
+            Policy::Older(t) => vec![],
```
Comment (sanket1729) [inline src/policy/semantic.rs]: The unused variable is introduced in the first commit and later fixed in another commit. The first commit should itself fix the unused variable.

**Review by sanket1729 — COMMENTED** [review]
Some general comments about the PR:

In general, usually, PRs are rebased on top of the latest master to avoid conflicts. Looks like you are merging the master branch onto the PR branch. 1. Read https://www.atlassian.com/git/tutorials/merging-vs-rebasing for starters. 
2. All commits in the PR should compile and pass the test. This helps tools like `git bisect` for easier debugging. The second commit in this PR 
37a2ad28636fb4694672db88c0efebab393f8375 has incorrect tests which are later fixed in a follow commit. 
3. The unused variable was introduced in the first commit and later fixed in the third commit. We should just not introduce it in the first commit.

**Comment by sanket1729** [comment]
The CI is broken right now, once #250 is fixed. I will add CI enforcement.

**Comment by SarcasticNastik** [comment]
Thanks for the references. I've updated my master with the commits as suggested and have rebase-d it.

**Review by sanket1729 — APPROVED** [review]
tACK bfdc790ae0ce1baac8cd8ea1f04d55c34175f4dc.

File: src/policy/semantic.rs [hunk 2]
```diff
@@ -499,6 +499,34 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
         ret
     }
 
+    /// Helper function for recursion in `absolute timelocks`
+    pub fn real_absolute_timelocks(&self) -> Vec<u32> {
```
Comment (darosior) [inline src/policy/semantic.rs:503]: nit: if this is a helper function, we'd rather not make it public?

**Review by darosior — COMMENTED** [review]
ACK bfdc790ae0ce1baac8cd8ea1f04d55c34175f4dc -- modulo the API nit

**Comment by darosior** [comment]
ACK 0295e7f5d12a2061f1fd2286cefec592dff656a19f55f4f01305d6aa56630880ce

It's be better for 1091d07 to be squashed in 0295e7f5d12a2061f1fd2286cefec592dff656a19f55f4f01305d6aa56630880ce in order to keep the changes atomic, up to the maintainers tho and that can be considered a nit :)

**Comment by sanket1729** [comment]
@SarcasticNastik, normally I wouldn't be so picky, but it would be a good exercise for you to try. Can you please make the change that Darosior requested?

**Comment by SarcasticNastik** [comment]
I've made the commit changes suggested by @darosior and squashed them into 925ae34.

File: src/policy/semantic.rs (hunk 2 above)
Comment (sanket1729) [reply src/policy/semantic.rs]: Ideally, this should be been fixed in cac78151935e83a0cd0eb58da0cca175e8e4ba4e and not in the next commit, but it' okay :)

**Review by sanket1729 — APPROVED** [review]
ACK 925ae345c51769c9ff0da7908f26f90866cf5c52

**Pull request merged**